Framework/DataHandling/src/LoadShape.cpp

```diff
@@ -0,0 +1,214 @@
+#include "MantidDataHandling/LoadShape.h"
+
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/InstrumentValidator.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidAPI/Sample.h"
+
+#include "MantidGeometry/Instrument.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/EnabledWhenProperty.h"
+#include "MantidKernel/Exception.h"
+
+#include <Poco/File.h>
+
+namespace Mantid {
+namespace DataHandling {
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(LoadShape)
+
+using namespace Kernel;
+using namespace API;
+using namespace Geometry;
+
+void LoadShape::init() {
+  auto wsValidator = boost::make_shared<CompositeValidator>();
+  wsValidator->add<API::InstrumentValidator>();
+
+  // input workspace
+  declareProperty(
+      make_unique<WorkspaceProperty<>>("InputWorkspace", "", Direction::Input,
+                                       wsValidator),
+      "The name of the workspace containing the instrument to add the shape");
+
+  // shape file
+  const std::vector<std::string> extensions{".stl"};
+  declareProperty(
+      make_unique<FileProperty>("Filename", "", FileProperty::Load, extensions),
+      "The name of the file containing the shape. ");
+
+  // Output workspace
+  declareProperty(make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
+                                                   Direction::Output),
+                  "The name of the workspace that will be same as "
```

Comment:
I don't think you are enforcing that, and I don't see that you should or can. Not a good idea to manipulate the ADS within an algorithm execution method.

Also, since this algorithm is NOT modifying the data, either it should clone the instrument from the first workspace (to leave it intact), or the workspace should be specified as `InOut` to indicate that the input geometry is going to be modified.